engine/swarm/services.md

```diff
@@ -70,6 +70,128 @@ $ docker service create --name helloworld alpine ping docker.com
 9uk4639qpg7npwf3fn2aasksr
 ```
 
+### Specify the image version the service should use
+
+When you create a service without specifying any details about the version of
+the image to use, the service uses the latest version available in Docker Hub
+or your registry. You can force the service to use a specific version of
+the image in a few different ways, depending on your desired outcome.
+
+An image version can be expressed in several different ways:
+
+- If you don't specify a version at all, by convention the image's `latest` tag
+  is resolved to a _digest_, which is a unique identifier for a version of the
+  image. The following two commands are equivalent:
+  ```bash
+  $ docker service create --name="myservice" ubuntu
+
+  $ docker service create --name="myservice" ubuntu:latest
+  ```
+
+- If you specify a different tag, Docker resolves that tag to a digest, and uses
+  that version.
+  ```bash
+  $ docker service create --name="myservice" ubuntu:16:04
+  ```
+
+- If you specify a digest directly, that exact version of the image is always
+  used.
+  ```bash
+  $ docker service create --name="myservice" ubuntu:16.04@sha256:35bc48a1ca97c3971611dc4662d08d131869daa692acb281c7e9e052924e38b1
+  ```
+
+**Note: Using `latest`** There is no guarantee that the version tagged
+with `latest` is actually the newest or most up-to-date version. In general, it
+is recommended that you choose a specific version of an image rather than
+relying on `latest`.
+
+You can specify any tag or digest which has been pushed to Docker Hub or your
+private registry. When you create a service, the swarm manager resolves the
+image's tag to the specific digest it points to at the time of service creation.
+If swarm manager is able to resolve the tag to a digest, all tasks for that
+service will use that specific digest unless the service is explicitly updated.
+
+If the swarm manager is not able to resolve the tag to a digest, each worker
+node is responsible for resolving the tag to a digest, and different nodes may
+use different versions of the image. If this happens, a warning like the
+following will be logged, substituting the placeholders for real information.
+
+```none
+unable to pin image <image_name> to digest: <reason>
+```
+
+To see an image's current digest, issue the command `docker inspect
+<image>:<tag>` and look for the `RepoDigests` line. The following is the current
+digest for `ubuntu:latest` at the time this content was written. The output is
+truncated for clarity.
+
+```bash
+$ docker inspect ubuntu:latest
+```
+
+```json
+"RepoDigests": [
+    "ubuntu@sha256:35bc48a1ca97c3971611dc4662d08d131869daa692acb281c7e9e052924e38b1"
+],
+```
+
+After you create a service, its image is never updated unless you explicitly run
+`docker service update` with the `--image` flag as described below. Other update
```

Comment:
Again, this is only if the manager is able to resolve the tag to digest. Otherwise this is not guaranteed.